### ISSUE TITLE
Host ASP.NET Core in a Windows Service

### ISSUE THREAD

**Issue by jkoplo** [dotnet/AspNetCore.Docs#20159]
I'm trying to follow this example to turn the simple Blazor Client Service Worker template into a service. I created a new user and made a powershell script for the block to set the user permissions and create the service. However I keep getting an error on the following line:

```powershell
$aclRuleArgs = {DOMAIN OR COMPUTER NAME\USER}, "Read,Write,ReadAndExecute", "ContainerInherit,ObjectInherit", "None", "Allow"
```

First of all, it seems like `{DOMAIN OR COMPUTER NAME\USER}` should be in quotes: `"{DOMAIN OR COMPUTER NAME\USER}"`

But even with that change, I get a `Missing argument in parameter list` right after the user. I've done some Googling, but nothing obvious is coming up.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b0bd8cb8-7e28-8c4d-a550-3aea8f3ab4dd
* Version Independent ID: 4b60d24e-ef05-15e5-10b4-483094081bf3
* Content: [Host ASP.NET Core in a Windows Service](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/windows-service?view=aspnetcore-3.1&tabs=visual-studio)
* Content Source: [aspnetcore/host-and-deploy/windows-service.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/host-and-deploy/windows-service.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @jkoplo ... I don't have any information on this hosting scenario at this time. I'm going to take a look at this scenario when I get to https://github.com/dotnet/AspNetCore.Docs/issues/19294. However, with .NET 5 upon us and the holidays hitting after that, it might be early next year before I reach this issue (and that other one). Leave this issue open, and I'll work them together. Your best bet in the meantime is to inquire with community support channels ...

* [Stack Overflow](https://stackoverflow.com/questions/tagged/blazor)
* [ASP.NET Core Slack](http://tattoocoder.com/aspnet-slack-sign-up/)
* [ASP.NET Gitter](https://gitter.im/aspnet/Blazor)

I'll get to this as soon as I can. If you work out the solution, please post back here. Leave the issue open tho ... it will close automatically when a PR merges later.

**Comment by jkoplo** [comment]
I've made some progress since my post. I created the service without a user and then manually modified the properties on the service to Log on as my personal (local admin) account. This was enough to allow the service to run and the Blazor page to be loaded from `localhost:5001`. 

This confirms that I have a permissions issue, but I'm not yet sure if the permissions are related to file access (static Blazor assets) or to opening the port on localhost. I'll test with some manual user creation and permissions modifications.

**Comment by jkoplo** [comment]
The creation of a separate service user and the permissions associated with it seems to have been a red-herring (for my purposes). Instead I created the service and omitted the credentials:
```
New-Service -Name {SERVICE NAME} -BinaryPathName {EXE FILE PATH} -Description "{DESCRIPTION}" -DisplayName "{DISPLAY NAME}" -StartupType Automatic
```

Everything runs fine under the default `Local System` credentials. What we tripping me up is that my app is not using tls/authentication and my web browser was defaulting to `https` which wasn't apparent in the address bar.

**Comment by guardrex** [comment]
I see ... so it's turned out not to be a Blazor-specific problem then as far as I can tell. If that's correct, then we can close this and I'll only need to take a look at https://github.com/dotnet/AspNetCore.Docs/issues/19294.

**Comment by jkoplo** [comment]
Yeah, not Blazor specific, but the powershell commands definitely don't work (on my machine, in PS7).

As for #19294, I didn't have to make any changes to the template for hosting Blazor in a Windows service other than adding the package and the UseWindowsService() call.

**Comment by guardrex** [comment]
Ok ... I'll focus on the PS command aspect when I reach this issue. I wrote "PowerShell 6" so many times (each command) because what the topic shows definitely doesn't work with PS 5. The topic was written when PS 6 was young, so I had to make it very clear to readers that the commands wouldn't work if they were on PS 5. I'll see about PS 7 when I work on this.

**Comment by guardrex** [comment]
I've reached this for work now. I'll investigate a few reported errors as I work through the updates. Updates include removing the 2.x pieces and adding bits for 5.x and 6.x. I'll deal with PS script updates. Note that the PS updates move all guidance to working PS 7 commands. *We don't document old tooling.*

**Comment by guardrex** [comment]
Sorry folks! Change of plans. One of the docs team members will take care of the general updates first (3/4 open issues), then I'll come in later and address the Blazor-specific issue (**THIS ONE**) later.
